Assets/ScreenSwipe/ScreenSwipe.cs

```diff
@@ -8,771 +8,818 @@
+        {
+            content.anchoredPosition = position;
+            UpdateContentBounds();
+        }
+    }
+
+    /// <summary>
+    /// Updates the bounds of the scroll view content
+    /// </summary>
+	private void UpdateContentBounds()
+    {
+        viewBounds = new Bounds(rectTransform.rect.center, rectTransform.rect.size);
+        contentBounds = GetContentBounds();
+
+        // Make sure content bounds are at least as large as view by adding padding if not.
+        // One might think at first that if the content is smaller than the view, scrolling should be allowed.
+        // However, that's not how scroll views normally work.
+        // Scrolling is *only* possible when content is *larger* than view.
+        // We use the pivot of the content rect to decide in which directions the content bounds should be expanded.
+        // E.g. if pivot is at top, bounds are expanded downwards.
+        // This also works nicely when ContentSizeFitter is used on the content.
+        Vector3 contentSize = contentBounds.size;
+        Vector3 contentPos = contentBounds.center;
+        Vector3 excess = viewBounds.size - contentSize;
+        if (excess.x > 0)
+        {
+            contentPos.x -= excess.x * (content.pivot.x - 0.5f);
+            contentSize.x = viewBounds.size.x;
+        }
+
+        contentBounds.size = contentSize;
+        contentBounds.center = contentPos;
+    }
+
+    /// <summary>
+    /// Gets the bounds of the scroll view content
+    /// </summary>
+    /// <returns>Content Bounds</returns>
+	private Bounds GetContentBounds()
+    {
+        var vMin = new Vector3(float.MaxValue, float.MaxValue, float.MaxValue);
+        var vMax = new Vector3(float.MinValue, float.MinValue, float.MinValue);
+
+        var toLocal = rectTransform.worldToLocalMatrix;
+
+        Vector3[] m_Corners = new Vector3[4];
+        content.GetWorldCorners(m_Corners);
+        for (int j = 0; j < 4; j++)
+        {
+            Vector3 v = toLocal.MultiplyPoint3x4(m_Corners[j]);
+            vMin = Vector3.Min(v, vMin);
+            vMax = Vector3.Max(v, vMax);
+        }
+
+        var bounds = new Bounds(vMin, Vector3.zero);
+        bounds.Encapsulate(vMax);
+        return bounds;
+    }
+
+    /// <summary>
+    /// Offset to get content into place in the view.
+    /// </summary>
+    /// <param name="pointerDelta">Pointer delta</param>
+    /// <returns>Offset to get content into place in the view.</returns>
+	private Vector2 CalculateOffset(Vector2 pointerDelta)
+    {
+        Vector2 offset = Vector2.zero;
+
+        Vector2 min = contentBounds.min;
+        Vector2 max = contentBounds.max;
+
+        min.x += pointerDelta.x;
+        max.x += pointerDelta.x;
+        if (min.x > viewBounds.min.x)
+            offset.x = viewBounds.min.x - min.x;
+        else if (max.x < viewBounds.max.x)
+            offset.x = viewBounds.max.x - max.x;
+
+        return offset;
+    }
+
+    /// <summary>
+    /// 
```

Comment:
I have no idea what this method is, or what it is calculating.

If this could be expanded to maintain readability and help other devs understand what it going on, that would be great

Reply:
I got the code from [here ](https://bitbucket.org/Unity-Technologies/ui/src/0651862509331da4e85f519de88c99d0529493a5/UnityEngine.UI/UI/Core/ScrollRect.cs?at=2018.3%2Fstaging#lines-1321) it looks like they have made some changes to core of how ScrollRect works. I think it might actually be better to extend from ScrollRect rather than just use its code like that. I can't remember why I did it this way, it was a long time ago. I will have a look at this project again tonight

Reply:
Ty, ill leave this link here if i need it again in future

https://bitbucket.org/Unity-Technologies/ui/src/0651862509331da4e85f519de88c99d0529493a5/UnityEngine.UI/UI/Core/ScrollRect.cs?at=2018.3%2Fstaging&fileviewer=file-view-default#lines-1026